### ISSUE TITLE
Magento 2.1.6 CE  "We Can't Place The Order" error

### ISSUE THREAD

**Issue by The-Krang** [magento/magento2#9455]
**My setup:**

2.0.7 CE was installed using Softaculous through CPanel from the web server.  Using the Luma theme.

I upgraded from 2.0.7 CE to 2.1.6 CE under System > Web Setup Wizard > System Upgrade.

With PHP Version: 5.6.30, Apache and Linux Web server, phpMyAdmin  version 4.0.10.18

Everything was working fine before the upgrade.

I have a success.phtml module that was disabled before the upgrade took place, and is still disabled.

The order success page is not displayed.

The site has only plain downloadable items, non-configurable, and all of them are In Stock and Enabled
.
PayPal Express Checkout is the only configured payment option, with the exact settings as before the upgrade.

PayPal Debug mode is on.

PayPal Skip Order Review Step = yes.

PayPal Payment Action = Sale

In sandbox mode in paypal, testing with a PayPal generated fake credit card number etc., with the site in developement mode.

The orders show up in the Admin at Sales > Orders, but are replaced by the most recent one.  They will only increment beyond 000000132 if the order is placed in the Admin.

On the "Review Order" page, it says "We can't place the order.".  Clicking the "Place Order" button at the bottom right of the page just shows the same page with the same error.

I have NOT made any changes to the database tables before or after the upgrade.

Your help and suggestions are much appreciated.

-------------------------------------------------------------------------------------------------------------------------
**Steps to reproduce:**

Add item(s)  to the shopping cart.
Proceed to shopping cart page.
Proceed to checkout.
The customer logs into his/her account on my store.
Proceed to the payment page, select Pay With Paypal as payment option.
Press Continue to Paypal.
The customer is redirected to the Paypal page.
Enter payment and other information into PayPal's page.
Confirm the order with the "Pay Now" button.

**Expected result:**

The customer is redirected to order success page which has the order number.
The customer receives an order confirmation e-mail.

**Actual result:**

The customer is  taken to the "Review Order" page, where it says in red letters,  "We can't place the order.".  Clicking the "Place Order" button at the bottom right of the page just shows the same page with the same error.

The order is recorded in Admin.
In the Admin > Sales > Orders > order number view, it reads, "Order # 000000132 (The order confirmation email is not sent)", **"Order Status	Suspected Fraud"**.

------------------------------------------------
**The main problem in the debug log appears to be:**

_main.CRITICAL: PayPal NVP gateway errors: A successful transaction has already 
been completed for this token (#10415: Transaction refused because of an 
invalid argument. See additional error messages for details). Correlation ID: 
219dd382778ee. Version: 72.0. [] []
[2017-04-18 20:39:19] main.CRITICAL: 
Exception message: PayPal gateway has rejected request. A successful 
transaction has already been completed for this token (#10415: Transaction 
refused because of an invalid argument._

**The main problem in the system log appears to be:**

_main.CRITICAL: Exception message: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '000000132-1' for key 'SALES_ORDER_INCREMENT_ID_STORE_ID', query was: INSERT INTO `e_sales_order_

Here are the debug and system logs:

[Debug.txt](https://github.com/magento/magento2/files/965702/Debug.txt)
[System.txt](https://github.com/magento/magento2/files/965703/System.txt)

**Comment by YPyltiai** [comment]
Hello @The-Krang,

Thank you for posting this. I've created an internal ticket to investigate the issue: MAGETWO-68949

**Comment by antoninobonumore** [comment]
We get the same identical issue

**Comment by s00071609** [comment]
Same here this issue has been there forever in magento 2. No fix yet or solution. DOes anyone know why this is being carried over from magento 2 to mangento 2.1.7?

**Comment by YevSent** [comment]
Hi, @The-Krang, the fix merged into the develop branch.

**Comment by sladjansimic** [comment]
Hi,

I had the error "We can't place the order" on Magento 2.1.7, which was fixed today by turning off GTID_CONSISTENCY. Now I am able to process orders using PayPal.

system.log error:
main.CRITICAL: Exception message: SQLSTATE[HY000]: General error: 1785 When @@GLOBAL.ENFORCE_GTID_CONSISTENCY = 1, updates to non-transactional tables can only be done in either autocommitted statements or single-statement transactions, and never in the same statement as updates to transactional tables., query was: INSERT INTO `sequence_invoice_1` () VALUES ()

Fix:
change /etc/my.cnf to turn off GTID_CONSISTENCY
gtid_mode = OFF
comment out enforce-gtid-consistency

I hope this helps.

**Comment by s00071609** [comment]
@sladjansimic 
Did it fix this error:


[2017-07-12 23:11:31] main.CRITICAL: Exception: Wrong order ID: "INV2-HCZX-TDHA-EA8F-3TZQ". in /home/couponc1/public_html/vendor/magento/module-paypal/Model/Ipn.php:140
Stack trace:
#0 /home/couponc1/public_html/vendor/magento/module-paypal/Model/Ipn.php(106): Magento\Paypal\Model\Ipn->_getOrder()
#1 /home/couponc1/public_html/vendor/magento/module-paypal/Model/Ipn.php(87): Magento\Paypal\Model\Ipn->_getConfig()
#2 /home/couponc1/public_html/vendor/magento/module-paypal/Controller/Ipn/Index.php(56): Magento\Paypal\Model\Ipn->processIpnRequest()
#3 /home/couponc1/public_html/vendor/magento/framework/App/Action/Action.php(102): Magento\Paypal\Controller\Ipn\Index->execute()
#4 /home/couponc1/public_html/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\App\Action\Action->dispatch(Object(Magento\Framework\App\Request\Http))
#5 /home/couponc1/public_html/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Paypal\Controller\Ipn\Index\Interceptor->___callParent('dispatch', Array)
#6 /home/couponc1/public_html/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Paypal\\...', 'dispatch', Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Array, 'contextPlugin')
#7 /home/couponc1/public_html/vendor/magento/module-store/App/Action/Plugin/Context.php(106): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#8 /home/couponc1/public_html/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\Action\Plugin\Context->aroundDispatch(Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#9 /home/couponc1/public_html/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Paypal\\...', 'dispatch', Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Array, 'customer-app-ac...')
#10 /home/couponc1/public_html/vendor/magento/module-customer/Model/App/Action/ContextPlugin.php(61): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#11 /home/couponc1/public_html/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Customer\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#12 /home/couponc1/public_html/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Paypal\\...', 'dispatch', Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Array, 'storeCheck')
#13 /home/couponc1/public_html/vendor/magento/module-store/App/Action/Plugin/StoreCheck.php(44): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#14 /home/couponc1/public_html/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\Action\Plugin\StoreCheck->aroundDispatch(Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#15 /home/couponc1/public_html/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Paypal\\...', 'dispatch', Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Array, 'weee-app-action...')
#16 /home/couponc1/public_html/vendor/magento/module-weee/Model/App/Action/ContextPlugin.php(112): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#17 /home/couponc1/public_html/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Weee\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#18 /home/couponc1/public_html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Paypal\\...', 'dispatch', Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Array, 'tax-app-action-...')
#19 /home/couponc1/public_html/vendor/magento/module-tax/Model/App/Action/ContextPlugin.php(91): Magento\Paypal\Controller\Ipn\Index\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#20 /home/couponc1/public_html/vendor/magento/framework/Interception/Interceptor.php(142): Magento\Tax\Model\App\Action\ContextPlugin->aroundDispatch(Object(Magento\Paypal\Controller\Ipn\Index\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#21 /home/couponc1/public_html/var/generation/Magento/Paypal/Controller/Ipn/Index/Interceptor.php(26): Magento\Paypal\Controller\Ipn\Index\Interceptor->___callPlugins('dispatch', Array, Array)
#22 /home/couponc1/public_html/vendor/magento/framework/App/FrontController.php(55): Magento\Paypal\Controller\Ipn\Index\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#23 /home/couponc1/public_html/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\App\FrontController->dispatch(Object(Magento\Framework\App\Request\Http))
#24 /home/couponc1/public_html/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\App\FrontController\Interceptor->___callParent('dispatch', Array)
#25 /home/couponc1/public_html/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'requestPreproce...')
#26 /home/couponc1/public_html/vendor/magento/module-store/App/FrontController/Plugin/RequestPreprocessor.php(94): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#27 /home/couponc1/public_html/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Store\App\FrontController\Plugin\RequestPreprocessor->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#28 /home/couponc1/public_html/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'install')
#29 /home/couponc1/public_html/vendor/magento/framework/Module/Plugin/DbStatusValidator.php(69): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#30 /home/couponc1/public_html/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Framework\Module\Plugin\DbStatusValidator->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#31 /home/couponc1/public_html/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'front-controlle...')
#32 /home/couponc1/public_html/app/code/Litespeed/Litemage/Model/App/FrontController/LitemagePlugin.php(71): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#33 /home/couponc1/public_html/vendor/magento/framework/Interception/Chain/Chain.php(67): Litespeed\Litemage\Model\App\FrontController\LitemagePlugin->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#34 /home/couponc1/public_html/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'front-controlle...')
#35 /home/couponc1/public_html/vendor/magento/module-page-cache/Model/App/FrontController/VarnishPlugin.php(55): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#36 /home/couponc1/public_html/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\PageCache\Model\App\FrontController\VarnishPlugin->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#37 /home/couponc1/public_html/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'front-controlle...')
#38 /home/couponc1/public_html/vendor/magento/module-page-cache/Model/App/FrontController/BuiltinPlugin.php(68): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#39 /home/couponc1/public_html/vendor/magento/framework/Interception/Interceptor.php(142): Magento\PageCache\Model\App\FrontController\BuiltinPlugin->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#40 /home/couponc1/public_html/var/generation/Magento/Framework/App/FrontController/Interceptor.php(26): Magento\Framework\App\FrontController\Interceptor->___callPlugins('dispatch', Array, Array)
#41 /home/couponc1/public_html/vendor/magento/framework/App/Http.php(135): Magento\Framework\App\FrontController\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#42 /home/couponc1/public_html/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#43 /home/couponc1/public_html/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#44 {main} [] []

**Comment by angelo983** [comment]
Putting gtid_mode = OFF seems to me an invasive way, someone of Magento Dev Team can confirm that no other fix can be applied?

**Comment by jer23jk** [comment]
Any solution for this bug?

Magento 2.1.7 > Order Variable product with custom option Date > Pay with Paypal

Transaction in successful, but after redirecting from paypal there is a red statement We can't place the order.


![image](https://user-images.githubusercontent.com/24847340/28848676-2a22e9f0-7714-11e7-8f60-fd41454f52eb.png)

**Comment by ghost** [comment]
We have the same problem with paypal.
Here the last lines of the debug log:
```
[2017-08-03 11:45:36] main.DEBUG: array (
  'url' => 'https://api-3t.paypal.com/nvp',
  'DoExpressCheckoutPayment' => 
  array (
    'TOKEN' => 'EC-2PA2142313410181M',
    'PAYERID' => '<ID>',
    'PAYMENTACTION' => 'Sale',
    'AMT' => '25.90',
    'CURRENCYCODE' => 'EUR',
    'BUTTONSOURCE' => 'Magento_Cart_Community',
    'NOTIFYURL' => 'https://<url of the shop>/paypal/ipn/',
    'RETURNFMFDETAILS' => 1,
    'SHIPPINGAMT' => '4.11',
    'ITEMAMT' => '17.65',
    'TAXAMT' => '4.14',
    'L_NUMBER0' => NULL,
    'L_NAME0' => 'Baumwoll Gästetuch - weiß mit farbigen Streifen',
    'L_QTY0' => 1,
    'L_AMT0' => '5.88',
    'L_NUMBER1' => NULL,
    'L_NAME1' => 'Baumwoll Gästetuch - weiß mit farbigen Streifen x2',
    'L_QTY1' => 1,
    'L_AMT1' => '11.77',

    ... (private informations of the customer) ...
    
    'ADDROVERRIDE' => 1,
    'METHOD' => 'DoExpressCheckoutPayment',
    'VERSION' => '72.0',
    'USER' => '****',
    'PWD' => '****',
    'SIGNATURE' => '****',
  ),
  'response' => 
  array (
    'TOKEN' => 'EC-2PA2142313410181M',
    'SUCCESSPAGEREDIRECTREQUESTED' => 'false',
    'TIMESTAMP' => '2017-08-03T11:45:35Z',
    'CORRELATIONID' => '77706e77b99ab',
    'ACK' => 'Failure',
    'VERSION' => '72.0',
    'BUILD' => '000000',
    'L_ERRORCODE0' => '10415',
    'L_SHORTMESSAGE0' => 'Transaction refused because of an invalid argument. See additional error messages for details.',
    'L_LONGMESSAGE0' => 'A successful transaction has already been completed for this token.',
    'L_SEVERITYCODE0' => 'Error',
  ),
) {"is_exception":false} []
[2017-08-03 11:46:52] main.DEBUG: array (
  'url' => 'https://api-3t.paypal.com/nvp',
  'DoExpressCheckoutPayment' => 
  array (
    'TOKEN' => 'EC-2PA2142313410181M',
    'PAYERID' => '<ID>',
    'PAYMENTACTION' => 'Sale',
    'AMT' => '25.90',
    'CURRENCYCODE' => 'EUR',
    'BUTTONSOURCE' => 'Magento_Cart_Community',
    'NOTIFYURL' => 'https://<url of the shop>/paypal/ipn/',
    'RETURNFMFDETAILS' => 1,
    'SHIPPINGAMT' => '4.11',
    'ITEMAMT' => '17.65',
    'TAXAMT' => '4.14',
    'L_NUMBER0' => NULL,
    'L_NAME0' => 'Baumwoll Gästetuch - weiß mit farbigen Streifen',
    'L_QTY0' => 1,
    'L_AMT0' => '5.88',
    'L_NUMBER1' => NULL,
    'L_NAME1' => 'Baumwoll Gästetuch - weiß mit farbigen Streifen x2',
    'L_QTY1' => 1,
    'L_AMT1' => '11.77',

    ... (private informations of the customer) ...
    
    'ADDROVERRIDE' => 1,
    'METHOD' => 'DoExpressCheckoutPayment',
    'VERSION' => '72.0',
    'USER' => '****',
    'PWD' => '****',
    'SIGNATURE' => '****',
  ),
  'response' => 
  array (
    'TOKEN' => '<TOKEN>',
    'SUCCESSPAGEREDIRECTREQUESTED' => 'false',
    'TIMESTAMP' => '2017-08-03T11:46:52Z',
    'CORRELATIONID' => '<ID>',
    'ACK' => 'Failure',
    'VERSION' => '72.0',
    'BUILD' => '000000',
    'L_ERRORCODE0' => '10415',
    'L_SHORTMESSAGE0' => 'Transaction refused because of an invalid argument. See additional error messages for details.',
    'L_LONGMESSAGE0' => 'A successful transaction has already been completed for this token.',
    'L_SEVERITYCODE0' => 'Error',
  ),
) {"is_exception":false} []
```

Thank you very much for help!

**Comment by andidhouse** [comment]
wow - we just lost about 50 orders on one day before we looked at the logs - this is way more then a problem! it is unbelievably how buggy magento2 is still after over 1.5 years released.

**Comment by s00071609** [comment]
You would have not even noticed other lost orders because no one informed there was an issue. Its quite surprising- the most important part of website that is payment, has such a bug that has been reported so widely and there is no fix for such a long time

**Comment by andidhouse** [comment]
The bad programming of M2 is a shame for this company. Payment issues which are not solved for such a time period is complete nonsense - but we did not expect anything else from magento side.

**Comment by jer23jk** [comment]
Will there be fix anytime soon?

**Comment by aceart** [comment]
Whats the current status on this issue?
Is it fixed on a newer version?
Thank you

**Comment by pepijnblom** [comment]
Magento hasn't supported GTID replication since v1. Guess who uses GTID in their HA database setups? Amazon, Google, Azure, Rackspace, and the list goes on.

**Comment by chequille** [comment]
When there will be s solution of this problem?
Nobody can work with such buggy software

Jürgen

**Comment by metaphist** [comment]
Had the same error after returning from Paypal, while transaction was made, order was not and message appeared on order review page.

Problem turned out to be Magento_Signifyd module being disabled. Found an exception in /var/log/system.log, with paypal classes in the stack trace:

report.CRITICAL: Exception message: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'magento.signifyd_case' doesn't exist

Re-enabling the module (re-compile, etc) fixed the issue.

**Comment by ksz2013** [comment]
Will this be fixed in 2.1.x latest release?

Or just on 2.3?

Regards

**Comment by rusdragos** [comment]
We managed to apply a fix on Magento 2.1 and fix below issue.
"**A successful transaction has already been completed for this token.**"

https://support.weltpixel.com/hc/en-us/articles/360000608754-How-to-fix-PayPal-Magento-2-checkout-issues-transaction-refused-invalid-argument-token-error

Hope this helps.
